### DISCUSSION TITLE
winget aggresively now prompts to accept terms unnecessarily

### DISCUSSION THREAD

**Discussion by nascentt** [microsoft/winget-cli#1536] [Q&A] [answered]
# Brief description of your issue
Prompts "Do you agree to all the source agreements terms" even when doing a `winget upgrade` that has no action
there's no way to see which applications are out of date without answering yes to that prompt.  that prompt shouldn't even show on passive commands.

There also needs to be a way to accept the terms via config, or parameter to avoid breaking non-interactive use.

**Comment by denelon** [comment, marked as answer]
@nascentt, there is an `--accept-package-agreements` parameter for package level agreements.

There is an `--accept-source-agreements` parameter for source level agreements.

These are listed in help for the command `winget upgrade -?`

Link to Microsoft Docs for upgrade command: https://docs.microsoft.com/windows/package-manager/winget/upgrade

One you have accepted a source agreement, you shouldn't be prompted again unless the source agreement changes.

Can you confirm if you are being prompted after accepting the source level agreement?

**Comment by cbotsikas** [comment]
EDIT: had the same argument twice 😕 It's --accept-**source**-agreements and --accept-**package**-agreements

Is there a solution when the app is from ms store?
I've tried both ` --accept-source-agreements` and  `--accept-source-agreements` but I'm still getting a prompt
`winget install -e --silent --accept-source-agreements --accept-source-agreements --id 9PGZKJC81Q7J #cinebench`
```
The publisher requires that you view the above information and accept the agreements before installing.
Do you agree to the terms?
[Y] Yes  [N] No:
```

**Reply by denelon** [reply]
It looks like you accepted source agreements twice, I believe:

`winget install 9PGZKJC81Q7J --accept-source-agreements --accept-package-agreements`

should do what you are looking for.

**Reply by cbotsikas** [reply]
omg. thanks, I probably never did the second copy 😵‍💫 
`winget install -e --silent --accept-source-agreements --accept-package-agreements --id 9PGZKJC81Q7J`

**Reply by nemchik** [reply]
Relevant https://github.com/microsoft/winget-cli/issues/2680

**Comment by lethallunatic** [comment]
winget upgrade --all --accept-source-agreements --accept-package-agreements worked for me to upgrade all installed packages

**Comment by paradonym** [comment]
we need a permanent flag to accept every future package agreement. Similar to what "choco feature enable -n=allowGlobalConfirmation" does.
Basically for every option. 

Typing "winget upgrade --all --include-unknown --accept-source-agreements --accept-package-agreements" for a "Just do it"-prompt is brutal...

**Reply by denelon** [reply]
* #2404

**Reply by atoav** [reply]
As a Linux person this should just be a `-y` (stands for *yes*) option.
